Signal/src/ViewControllers/FingerprintViewScanController.m

```diff
@@ -0,0 +1,218 @@
+//
+//  Copyright (c) 2017 Open Whisper Systems. All rights reserved.
+//
+
+#import "FingerprintViewScanController.h"
+#import "Environment.h"
+#import "OWSContactsManager.h"
+#import "OWSQRCodeScanningViewController.h"
+#import "Signal-Swift.h"
+#import "UIColor+OWS.h"
+#import "UIFont+OWS.h"
+#import "UIUtil.h"
+#import "UIView+OWS.h"
+#import "UIViewController+CameraPermissions.h"
+#import <SignalServiceKit/OWSError.h>
+#import <SignalServiceKit/OWSFingerprint.h>
+#import <SignalServiceKit/OWSFingerprintBuilder.h>
+
+NS_ASSUME_NONNULL_BEGIN
+
+@interface FingerprintViewScanController () <OWSQRScannerDelegate>
+
+@property (nonatomic) TSStorageManager *storageManager;
+@property (nonatomic) OWSFingerprint *fingerprint;
+@property (nonatomic) NSString *contactName;
+@property (nonatomic) OWSQRCodeScanningViewController *qrScanningController;
+
+@end
+
+@implementation FingerprintViewScanController
+
+- (void)configureWithRecipientId:(NSString *)recipientId
+{
+    OWSAssert(recipientId.length > 0);
+
+    self.storageManager = [TSStorageManager sharedManager];
+
+    OWSContactsManager *contactsManager = [Environment getCurrent].contactsManager;
+    self.contactName = [contactsManager displayNameForPhoneIdentifier:recipientId];
+
+    OWSRecipientIdentity *_Nullable recipientIdentity =
+        [[OWSIdentityManager sharedManager] recipientIdentityForRecipientId:recipientId];
+    OWSAssert(recipientIdentity);
+
+    OWSFingerprintBuilder *builder =
+        [[OWSFingerprintBuilder alloc] initWithStorageManager:self.storageManager contactsManager:contactsManager];
+    self.fingerprint =
+        [builder fingerprintWithTheirSignalId:recipientId theirIdentityKey:recipientIdentity.identityKey];
+}
+
+- (void)loadView
+{
+    [super loadView];
+
+    self.title = NSLocalizedString(@"SCAN_QR_CODE_VIEW_TITLE", @"Title for the 'scan QR code' view.");
+
+    [self createViews];
+}
+
+- (void)createViews
+{
+    UIColor *darkGrey = [UIColor colorWithRGBHex:0x404040];
+
+    self.view.backgroundColor = [UIColor blackColor];
+
+    self.qrScanningController = [OWSQRCodeScanningViewController new];
```

Comment:
We could (and probably should) combine `OWSQRCodeScanningViewController` with this VC.  Pressed for time at the moment.